newsfragments/890.feature.rst

```diff
@@ -0,0 +1,4 @@
+The Trio scheduler can now be made deterministic by seeding the internal
+random.Random instance that is used to shuffle available tasks.
+This is not public API, but is important for some tools such as
+pytest-trio's `Hypothesis <https://hypothesis.readthedocs.io>`_ integration.
```

Comment:
The description here of how this works is a bit out of date, and it's also probably more detail than people need for a single-consumer API involving monkeypatching :-). How about just saying in general terms that the scheduler can now be made fully deterministic when running under hypothesis?

Also, in ReST links you almost always want two underscores: https://stackoverflow.com/questions/27420317/restructured-text-rst-http-links-underscore-vs-use